lib/Stampie/Mailer.php

```diff
@@ -50,6 +60,26 @@ public function getAdapter()
     }
 
     /**
+     * @return MessageFactory
+     */
+    private function getMessageFactory()
+    {
+        return $this->messageFactory ?: MessageFactoryDiscovery::find();
```

Comment:
Don't know the internals for httplug, but shouldn't it have a method to get the message factory from its adapter? As they are closely related.